Formula/bazel.rb

```diff
@@ -15,6 +16,9 @@ class Bazel < Formula
 
   def install
     ENV["EMBED_LABEL"] = "#{version}-homebrew"
+    # Bazel has it owns sandbox, ensure it works.
+    ENV.delete("SDKROOT")
+    ENV["CC"] = `xcrun -f #{ENV["CC"]}`.strip
```

Comment:
> > (btw, the environment SDKROOT is simply incorrect. I don't know who set it, I don't have it in my environment so it's coming from some part of Homebrew).
> It's not necessary to set it in all cases, but it's far from being incorrect. Homebrew sets it in certain circumstances and tools like xcrun and a few others respect it. Also many build systems either pick it up or at least don't tamper with it. Check out the xcrun(1) man page for details.

Sorry I wasn't clear. I meant the HOMEBREW_SDKROOT is correct on my machine but not the SDKROOT. I don't know where is the second one points and where it is set. The superenv gcc use the former.

> 
> > There is no way to tweak the action environment in Bazel (that would be against its philosophy).
> Forgive me, but not having spent time studying or using Bazel, I can't really comment on this and don't understand what the “action environment” is. But this decision feels unfortunate.

Environment of Bazel's actions. Basically, Bazel strips the environment to avoid environment having effect on caching and on the action determinism.

The only part the environment is perserved is when generating that configuration file for C/C++ compilation to figure out various thing about the environment, notably where the system includes are (cc -E).

> > Note that the previous version of bazel was directly coding the path to /usr/bin/gcc in its C++ configuration file […]
> That's even worse. You realize that GCC is not the only C/C++ compiler suite, can be installed in many different locations and under different names, and specifically gcc is not even the right frontend if you're targeting C++ code? (Sorry for this mini-rant.)

Agree on that one, this is not correct, that's why we changed it.

> 
> To be honest, I'm not really happy with the situation and don't feel like we have found a good solution yet. But I'll ask other maintainers to have a look. I don't want to waste too much of your time if it turns out that I'm being overly stubborn and others find your hack/solution acceptable.

I am open to any better solution but I don't see any. Note that the re-writting of options done by the Homebrew superenv cc might also conflict with the tuning of options done by Bazel itself.